packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -1610,9 +1611,12 @@ export class Bucket extends BucketBase {
     }
     Bucket.validateBucketName(bucketName);
 
-    const newUrlFormat = attrs.bucketWebsiteNewUrlFormat === undefined
-      ? false
-      : attrs.bucketWebsiteNewUrlFormat;
+    const oldUrlFormatRegions = new Set(regionsBefore(RULE_S3_WEBSITE_REGIONAL_SUBDOMAIN));
+
+    const newUrlFormat = !oldUrlFormatRegions.has(region) ||
+      (attrs.bucketWebsiteNewUrlFormat === undefined
+        ? false
+        : attrs.bucketWebsiteNewUrlFormat);
```

Comment:
Nice improvement! Though I reckon we might as well use the existing helper for this:

https://github.com/aws/aws-cdk/blob/85e27351c6246f5a06b3cbef3e2961e1c6359add/packages/@aws-cdk/region-info/lib/region-info.ts#L97

```ts
const region = regionInfo.RegionInfo.get('eu-west-1');
region.s3StaticWebsiteEndpoint; // s3-website-eu-west-1.amazonaws.com
```

Reply:
Ah nice, will try it!

Reply:
How would you use it? I tried to use `region.s3StaticWebsiteEndpoint` for the new format but some tests are failing because of the suffix `amazonaws.com` 🤔

Reply:
Which tests are failing?

Honestly, the current use of `stack.urlSuffix` feels wrong. It appears to assume the suffix based on the Stack's region not the Bucket's region.

Reply:
Alright, was not sure if usage of stack's suffix intended or not, but I agree that feels kind of wrong, will fix this with helper `s3StaticWebsiteEndpoint()`soon 👌